### PR TITLE
Fixes #15360 - UserData default should Patch the system and reboot

### PR DESCRIPTION
**Pull request by edestecd**

### CODE REVIEW

File: cloudinit/userdata_cloudinit.erb [hunk 1]
```diff
@@ -41,8 +41,19 @@ users:
...
 <%# Contact Foreman to confirm instance is built -%>
 phone_home:
- url: <%= foreman_url('built') %>
- post: []
- tries: 10
+  url: <%= foreman_url('built') %>
+  post: []
+  tries: 10
+
+<% unless @host.param_false?('reboot') -%>
```
Comment (domcleal) [inline cloudinit/userdata_cloudinit.erb]: I'm worried about this behaviour change, and wonder if the default should be not to upgrade/reboot and have it opt-in?

**Comment by domcleal** [comment]
Also documentation for these parameters would be great.

File: cloudinit/userdata_cloudinit.erb (hunk 1 above)
Comment (edestecd) [inline cloudinit/userdata_cloudinit.erb]: OK, I agree with that.  Seems like more will not want this in a cloud environment.

**Comment by edestecd** [comment]
Where would you like documentation?  At the top of the template, with a line for each param, like others?

**Comment by domcleal** [comment]
Yep, top of the template please.

**Comment by edestecd** [comment]
hmm other templates use dasherized params, while this one uses underscore.
Should we keep it underscore or try to make it consistent by switching the two existing params to dash?

**Comment by domcleal** [comment]
It's probably better to prefer dashes if that's what the majority use already.

**Comment by edestecd** [comment]
Other templates are using the param 'ssh_authorized_keys' and I have seen others with underscore like 'salt_master'.

Its a mixed bag at this point.  I'm not willing to change them all.   Seems like that could break for many that are already using them.

**Comment by edestecd** [comment]
@domcleal  I think I fixed everything

**Comment by domcleal** [comment]
Thanks @edestecd!

**Pull request merged**